test/hotspot/jtreg/compiler/c2/irTests/SubLNodeIdealizationTests.java

```diff
@@ -0,0 +1,227 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package compiler.c2.irTests;
+
+import jdk.test.lib.Asserts;
+import compiler.lib.ir_framework.*;
+
+/*
+ * @test
+ * @bug 8267265
+ * @summary Test that Ideal transformations of SubLNode* are being performed as expected.
+ * @library /test/lib /
+ * @run driver compiler.c2.irTests.SubLNodeIdealizationTests
+ */
+public class SubLNodeIdealizationTests {
+    public static void main(String[] args) {
+        TestFramework.run();
+    }
+
+    @Run(test = {"test1", "test2", "test3",
+                 "test4", "test5", "test6",
+                 "test7", "test8", "test9",
+                 "test10", "test11", "test12",
+                 "test13", "test14", "test15",
+                 "test16", "test17", "test18"})
+    public void runMethod() {
+        long a = RunInfo.getRandom().nextLong();
+        long b = RunInfo.getRandom().nextLong();
+        long c = RunInfo.getRandom().nextLong();
+
+        long min = Long.MIN_VALUE;
+        long max = Long.MAX_VALUE;
+
+        assertResult(0, 0, 0);
+        assertResult(a, b, c);
+        assertResult(min, min, min);
+        assertResult(max, max, max);
+    }
+
+    @DontCompile
+    public void assertResult(long a, long b, long c) {
+        Asserts.assertEQ(a - 1            , test1(a));
+        Asserts.assertEQ((a + 1) - b      , test2(a, b));
+        Asserts.assertEQ(a - (b + 2021)   , test3(a, b));
+        Asserts.assertEQ(a - (a + b)      , test4(a, b));
+        Asserts.assertEQ(a - (b + a)      , test5(a, b));
+        Asserts.assertEQ(0 - (a - b)      , test6(a, b));
+        Asserts.assertEQ(0 - (a + 2021)   , test7(a, b));
+        Asserts.assertEQ((a + b) - (a + c), test8(a, b, c));
+        Asserts.assertEQ((b + a) - (c + a), test9(a, b, c));
+        Asserts.assertEQ((b + a) - (a + c), test10(a, b, c));
+        Asserts.assertEQ((a + b) - (c + a), test11(a, b, c));
+        Asserts.assertEQ(a - (b - c)      , test12(a, b, c));
+        Asserts.assertEQ(0 - (a >> 63)    , test13(a));
+        Asserts.assertEQ(0 - (0 - a)      , test14(a));
+        Asserts.assertEQ(a*b - a*c        , test15(a, b, c));
+        Asserts.assertEQ(a*b - b*c        , test16(a, b, c));
+        Asserts.assertEQ(a*c - b*c        , test17(a, b, c));
+        Asserts.assertEQ(a*b - c*a        , test18(a, b, c));
```

Comment:
Some cases are missing compared to `SubI`. Did you omit them on purpose?

Reply:
Yes, this transformation is actually missing for Long. I had created this RFE to fix that: https://bugs.openjdk.java.net/browse/JDK-8272735